com.unity.render-pipelines.core/Editor/RenderPipelineConverter.cs

```diff
@@ -0,0 +1,196 @@
+
+        internal FailedItem GetFailedItemAtIndex(int index)
+        {
+            if (index < 0 || index > failedCount)
+                throw new ArgumentOutOfRangeException();
+            return m_FailedItems[index];
+        }
+
+        internal SuccessfulItem GetSuccessfulItemAtIndex(int index)
+        {
+            if (index < 0 || index > successfulCount)
+                throw new ArgumentOutOfRangeException();
+            return m_SuccessfulItems[index];
+        }
+
+
+
+        /// <summary>
+        /// These are the items that the converter needs to iterate over.
+        /// The UI will make this list of items only the ones that are ticked and should be converted.
+        /// </summary>
+        /// <param name="items">The items the converter will operate on.</param>
+        public RunConverterContext(List<ConverterItemInfo> items)
+        {
+            m_Items = items;
+
+            m_FailedItems = new List<FailedItem>();
+            m_SuccessfulItems = new List<SuccessfulItem>();
+        }
+
+        /// <summary>
+        /// Mark the converter item index as failed.
+        /// </summary>
+        /// <param name="index">The index that will be marked as failed.</param>
+        public void MarkFailed(int index)
+        {
+            MarkFailed(index, "Failed");
+        }
+
+        /// <summary>
+        /// Mark the converter item index as failed.
+        /// </summary>
+        /// <param name="index">The index that will be marked as failed.</param>
+        /// <param name="message">The message that will be shown for the failed item.</param>
+        public void MarkFailed(int index, string message)
+        {
+            m_FailedItems.Add(new FailedItem(){index = index, message = message});
+        }
+
+        /// <summary>
+        /// Mark the converter item index as successful.
+        /// </summary>
+        /// <param name="index">The index that will be marked as successful.</param>
+        public void MarkSuccessful(int index)
+        {
+            m_SuccessfulItems.Add(new SuccessfulItem(){index = index});
+        }
+    }
+
+    // Might need to change this name before making it public
+    public abstract class RenderPipelineConverter
+    {
+        /// <summary>
+        /// Name of the converter.
+        /// </summary>
+        public abstract string name { get; }
+
+        /// <summary>
+        /// The information when hovering over the converter.
+        /// </summary>
+        public abstract string info { get; }
+
+        /// <summary>
+        /// A check if the converter is enabled or not. Can be used to do a check if prerequisites are met to have it enabled or disabled.
+        /// </summary>
+        public virtual bool Enabled()
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// This method getting triggered when clicking the listview item in the UI.
+        /// </summary>
+        public virtual void OnClicked(int index)
```

Comment:
What is this used for from a "I'm writing a converter"-perspective?

Reply:
Can be used as pinging an asset, or selecting it shoving it in inspector. Or open up quality settings, etc.